docs/pages.mjs

```diff
@@ -1,47 +1,65 @@
-import path, { dirname } from "path";
-import { fileURLToPath } from "url";
-import fs from "fs";
-
-function getFiles(dir) {
-  return fs.readdirSync(dir).flatMap((item) => {
-    const currentPath = path.join(dir, item);
-    if (fs.statSync(currentPath).isDirectory()) {
-      return {
-        dir: true,
-        name: item,
-        path: currentPath,
-      };
-    }
+import { readdir, readFile, stat } from 'node:fs/promises'
+import { join } from 'node:path'
+import { fileURLToPath } from 'node:url'
 
-    return {
-      dir: false,
+/**
+ * Scans the directory for files and collects them
+ * @param {import("node:fs").PathLike} dir The directory to scan the files from
+ * @returns {Promise<Array<{ dir: boolean; name: string; path: string; }>>} The files in the directory
+ */
+async function getFiles(dir) {
+  /**
+   * @type Array<{ dir: boolean; name: string; path: string; }>
+   */
+  const result = []
+  const dirAsPathString = dir instanceof URL ? fileURLToPath(dir) : dir
+
+  const dirContents = await readdir(dir)
+
+  for (const item of dirContents) {
+    const currentPath = join(dirAsPathString, item)
+
+    const currentPathStat = await stat(currentPath)
+
+    result.push({
+      dir: currentPathStat.isDirectory(),
       name: item,
       path: currentPath,
-    };
-  });
+    })
+  }
+
+  return result
 }
 
-function asPage(item) {
+/**
+ * Get all items as page objects
```

Comment:
There is a long lasting debate in the world between developers whether inline code documentation is useful or not. In this case, I find the `@param` and `@returns` useful, because they actually help the developers and the IDE to know what's in input and output. Something typescript gives naturally but not JS.

However, would you say that the comments add anything useful here? 
- `Get all items as page objects`
- `The item to parse`
- etc

Reply:
That's fair. Everyone has their own view in the debate mentioned. Personally I just tend to let my documentation be written by GitHub CoPilot and be done with it heheh.. yeah those lines were also. Anyway I'll remove them.